architecture/core.rst

```diff
@@ -121,31 +124,30 @@ ARM_DEN0028A_SMC_Calling_Convention_ for details).
 **Entry and exit of Trusted OS**
 
 On entry and exit of Trusted OS each CPU is uses a separate entry stack and runs
-with IRQ and FIQ blocked. SMCs are categorised in two flavors: **fast** and
-**standard**.
+with IRQ and FIQ masked. SMCs are categorised in two flavors: **fast** and
+**yielding**.
 
     - For **fast** SMCs, optee_os will execute on the entry stack with IRQ/FIQ
-      blocked until the execution returns to normal world.
+      masked until the execution returns to normal world.
 
-    - For **standard** SMCs, optee_os will at some point execute the requested
-      service with interrupts unblocked. In order to handle interrupts, mainly
+    - For **yielding** SMCs, optee_os will at some point execute the requested
+      service with interrupts unmasked. In order to handle interrupts, mainly
       forwarding of foreign interrupts, optee_os assigns a trusted thread
       (`core/arch/arm/kernel/thread.c`_) to the SMC request. The trusted thread
       stores the execution context of the requested service. This context can be
       suspended and resumed as the requested service executes and is
       interrupted. The trusted thread is released only once the service
       execution returns with a completion status.
 
-      For **standard** SMCs, optee_os allocates or resumes a trusted thread then
-      unblock the IRQ/FIQ lines. When the optee_os needs to invoke the normal
-      world from a foreign interrupt or a remote service call, optee_os blocks
-      IRQ/FIQ and suspends the trusted thread. When suspending, optee_os gets
-      back to the entry stack.
-
-    - **Both** fast and standard SMC end on the entry stack with IRQ/FIQ blocked
-      and optee_os invokes the Monitor through a SMC to return to the normal
-      world.
+      For **yielding** SMCs, optee_os allocates or resumes a trusted thread
+      then unmasks the IRQ and FIQ lines. When the optee_os needs to invoke the
```

Comment:
`optee_os` here is a re-use, but I wonder if it wouldn't be better to replace it with something like "OP-TEE core" or so? Thoughts? (if you agree, then there are a few more locations in this section that needs to be replaced).

Reply:
When there is no ambiguity related to kernel vs. user space, I think `OP-TEE OS` is fine (and preferable).

Reply:
We have been using OP-TEE OS further down. I'm fine with that as well. It's just the `optee_os` (git name) that felt wrong.

Reply:
Good, then it's settled.